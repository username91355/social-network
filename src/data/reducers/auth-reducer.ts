import {IMe, serverAPI} from "../../api/api";
import {ThunkType} from "../redux";

//Action types
const SUCCESS_AUTH = 'socialNetwork/authReducer/SUCCESS_AUTH'
const FAILED_AUTH = 'socialNetwork/authReducer/FAILED_AUTH'

//State
const initialState = {
    isAuth: false,
    id: null,
    login: null,
    email: null,
}

//Reducer
export const authReducer = (state: IAuthState = initialState, action: TAuthActions) => {
    switch (action.type) {
        case SUCCESS_AUTH:
            return {
                ...state,
                isAuth: true,
                ...action.payload
            }
        case FAILED_AUTH:
            return {
                ...state,
                isAuth: false,
                id: null,
                login: null,
                email: null,
            }
        default:
            return state
    }
}

//Types
interface IAuthState {
    isAuth: boolean,
    id: number | null,
    login: string | null,
    email: string | null,
}

type TAuthActions = TSuccessAuth | TFailedAuth
type TSuccessAuth = ReturnType<typeof successAuth>
type TFailedAuth = ReturnType<typeof failedAuth>

//Action creators
const successAuth = (payload: IMe) => ({type: SUCCESS_AUTH, payload} as const)
const failedAuth = () => ({type: FAILED_AUTH} as const)

//Thunks
export const setAuth = (): ThunkType => async dispatch => {
    const result = await serverAPI.me()

    if (result.resultCode === 0) {
        dispatch(successAuth(result.data))
    } else {
        dispatch(failedAuth())
    }
}

export const login = (email: string, password: string, rememberMe: boolean): ThunkType => async dispatch => {
    const result = await serverAPI.login(email, password, rememberMe, false)

    if (result.resultCode === 0) {
        dispatch(setAuth())
    } else {
        dispatch(failedAuth())
    }
}

export const logout = (): ThunkType => async dispatch => {
    const result = await serverAPI.logout()

    if (result.resultCode === 0) {
        dispatch(failedAuth())
    }
}

