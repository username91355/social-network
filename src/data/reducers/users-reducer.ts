//Action Types
import {serverAPI, User} from "../../api/api";
import {ThunkType} from "../redux";

const SET_USERS = 'SET_USERS'

//Initial state
const initialState = {
    count: 10,
    page: 1,
    term: '',
    friend: null,
    users: [],
}

//Reducer
export const usersReducer = (state: IUsersState = initialState, action: any) => {
    switch (action.type) {
        case SET_USERS:
            return {
                ...state,
                users: action.users
            }
        default:
            return state
    }
}

//Types
interface IUsersState {
    count: number
    page: number
    term: string
    friend: null
    users: User[]
}

//Action creators
const setUsers = (users: User[]) => ({type: SET_USERS, users})

//Thunk

export const usersInitialization = (count: number, page: number, term: string, friend: boolean | null): ThunkType =>
    async dispatch => {
        const result = await serverAPI.getUser(count,page,term,friend)

        if(!result.error) {
            dispatch(setUsers(result.items))
        }
    }