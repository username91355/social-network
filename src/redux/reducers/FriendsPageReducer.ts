import { v1 } from "uuid"

//Action types
const ADD_FRIEND = 'ADD_FRIEND'
const REMOVE_FROM_FRIEND = 'REMOVE_FROM_FRIEND'
const SET_USERS = 'SET_USERS'

//Action Creators
const addFriendAC = (id: string) => {
    return {
        type: ADD_FRIEND,
        id: id
    }
}

const removeFromFriendAC = (id: string) => {
    return {
        type: REMOVE_FROM_FRIEND,
        id: id
    }
}

const setUsersAC = (users: any | never) => {
    return {
        type: SET_USERS,
        users: users
    }
}

let initialState: FriednsPageStateType = {

    allUsers: [
        
    ]
}

let FriednsPageReducer = (state = initialState, action: ActionCreatortype) => {
    switch (action.type) {
        
        case SET_USERS: {
            return {...state, 
                allUsers: [...state.allUsers, ...action.users.items]
            }
        }

        case ADD_FRIEND: {
            return {
                ...state,
                allUsers: state.allUsers.map(user => {
                    return (user.id === action.id) 
                    ? {...user, onFriends: true} 
                    : user
                })
            }
        }

        case REMOVE_FROM_FRIEND: {
            return {
                ...state,
                allUsers: state.allUsers.map(user => {
                    return (user.id === action.id) 
                    ? {...user, onFriends: false} 
                    : user
                })
            }
        }

        default: {
            return state
        }
    }

}

type FriednsPageStateType = {
    allUsers: Array<UserType>
}

type UserType = {
    id: string
    name: string
    onFriends: boolean
}

type ActionCreatortype = {
    type: string
    users?: any
    id?: string
    key?: string
}

export type FilterType = 'FILTER_ALL' | 'FILTER_FRIENDS' | 'FILTER_USERS'

// export type ServerUserType = {
//     followed: boolean
//     id: number
//     name: string
//     photos: ServerUserPhotoType
//     status?: null
//     uniqueUrlName?: null
// }

// export type ServerUserPhotoType = {
//     large: null
//     small: null
// }
    

export {addFriendAC, removeFromFriendAC, setUsersAC}
export default FriednsPageReducer