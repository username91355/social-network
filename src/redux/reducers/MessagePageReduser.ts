import { type } from "os"
import { v1 } from "uuid"

//Action types
const CHANGE_MESSAGE_TEXT_AREA = 'CHANGE_MESSAGE_TEXT_AREA'
const SEND_MESSAGE = 'SEND_MESSAGE'

//Action Creators
const sendMessageAC = (userId: string, value: string) => {
    return {
        type: SEND_MESSAGE,
        userId: userId,
        value: value
    }
}

const changeMessageTextAreaAC = (value: string) => {
    return {
        type: CHANGE_MESSAGE_TEXT_AREA,
        value: value
    }
}

let initialState: MessagesStateType  = {

    messages: [
        {id: v1(), userId: '5', incoming: true, text: 'Hello'},
        {id: v1(), userId: '5', incoming: true, text: 'You want go to walk?'},
        {id: v1(), userId: '0', incoming: false, text: 'HI. Yes i want'},
        {id: v1(), userId: '5', incoming: true, text: 'Great. See you soon.'},
        {id: v1(), userId: '0', incoming: false, text: 'Ok)'}
    ],

    newMessageTextArea: ''

}

let MessagesPageReducer = (state: MessagesStateType  = initialState, action: ActionType) => {
    switch (action.type) {
        case SEND_MESSAGE: {
            return {
                ...state,
                messages: [state.messages, {
                    id: v1(),
                    userId: action.userId,
                    incoming: action.userId !== '0',
                    text: action.value
                }]
            }
        }

        case CHANGE_MESSAGE_TEXT_AREA: {
            return {
                ...state,
                newMessageTextArea: action.value
            }
        }

        default: {
            return state
        }
    }
}

type MessagesStateType = {
    messages: Array<MessageType>
    newMessageTextArea: string
}

type MessageType = {
    id: string
    userId: string
    incoming: boolean
    text: string
}

type ActionType = {
    type: string
    id: string
    userId: string
    value: string
}

export { sendMessageAC, changeMessageTextAreaAC }
export default MessagesPageReducer