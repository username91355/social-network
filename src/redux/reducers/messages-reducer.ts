import {usersAPI} from "../../data/serverAPI";

//Action types
export const SET_DIALOGS = 'SET_DIALOGS'
export const CHANGE_NEW_MESSAGE_AREA = 'CHANGE_NEW_MESSAGE_AREA'
export const SEND_MESSAGE = 'SEND_MESSAGE'

//State
export const initialState = {

    dialogs: [],

    messages: {
        20345: [
            {id: 1, message: 'Hi!', outgoing: false},
            {id: 2, message: 'Hello!', outgoing: true},
            {id: 3, message: 'How are you?', outgoing: false},
            {id: 4, message: 'I`m fine. How ary you?', outgoing: true},
            {id: 5, message: 'I am OK.', outgoing: true},
            {id: 6, message: 'OK. Maybe go to the walk?', outgoing: false},
        ],
        20239: [
            {id: 1, message: 'Hello. What time do i need to arrive today?', outgoing: false},
            {id: 2, message: 'Hello. At 8pm', outgoing: true},
            {id: 3, message: 'Ok. Thx', outgoing: false}
        ],
        15190: [
            {id: 1, message: 'Hi. How are you?', outgoing: true},
            {id: 2, message: 'Good afternoon! I\'m on a business trip in Norway', outgoing: false},
        ],
        2: [
            {id: 1, message: 'Hi. Thanks for the help.', outgoing: true},
            {id: 2, message: 'Hi. You are always welcome!', outgoing: false},
            {id: 3, message: 'How are you?', outgoing: false},
        ]
    },
    newMessageText: ''
}

//Reducer
export const messagesReducer = (state: any = initialState, action: any) => {
    switch (action.type) {

        case SET_DIALOGS:
            return {
                ...state,
                dialogs: action.list
            }

        case CHANGE_NEW_MESSAGE_AREA:
            return {
                ...state,
                newMessageText: action.value
            }

        case SEND_MESSAGE: {

            const allMessageID = [...state.messages[action.userId].map((p:any) => p.id)]

            const generateID = (id: number): number => {
                return (allMessageID.some(i => +i === +id))
                    ? generateID(id + 1)
                    : id
            }

            return  {
                ...state,
                messages: {
                    ...state.messages,
                    [action.userId]: [
                        ...state.messages[action.userId],
                        {
                            id: generateID(Math.max(...allMessageID)),
                            message: state.newMessageText,
                            outgoing: true
                        }]
                },
                newMessageText: ''
            }
        }

        default:
            return state
    }
}

//Action creators
export const changeNewMessageArea = (value: string) => ({type: CHANGE_NEW_MESSAGE_AREA, value} as const)
export const sendMessage = (userId: number, message: string) => ({type: SEND_MESSAGE, userId, message} as const)
export const setDialogs = (list: any) => ({type: SET_DIALOGS, list} as const)

//Thunk
export const setDialogsTC = () => async (dispatch: any) => {
    const response = await usersAPI.getUsers(20, 1, '', true)
    dispatch(setDialogs(response.data.items))
}

//TYPSCRIPT
type initialStateType = {
    dialogs: Array<TUser>
    messages: Array<TMessage>
    newMessageText: string
}

export type TUser = {
    followed: boolean
    id: number
    name: string
    photos: {
        small: string | null
        large: string | null
    }
    status: string | null
    uniqueUrlName: string | null
}

type TMessage = {
    id: number
    message: string
    outgoing: boolean
}

export default messagesReducer