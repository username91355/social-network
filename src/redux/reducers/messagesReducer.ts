//Action type
const SEND_MESSAGE = 'SEND_MESSAGE'
const REMOVE_MESSAGE = 'REMOVE_MESSAGE'
const CHANGE_NEW_MESSAGE_AREA = 'CHANGE_NEW_MESSAGE_AREA'

//Action creator
export const sendMessage = () => ({type: SEND_MESSAGE} as const)
export const removeMessage = (id: number) => ({type: REMOVE_MESSAGE, id} as const)
export const changeNewMessageArea = (text: string) => ({type: CHANGE_NEW_MESSAGE_AREA, text} as const)

//Types
type MessagesStateType = {
    dialogs: Array<DialogType>
    messages: Array<MessageType>
    newMessageArea: string
}
export type DialogType = {
    id: number
    name: string
}
export type MessageType = {
    id: number
    message: string
}
type MessagesActionType = SendMessageType | RemoveMessageType | ChangeNewMessageArea
type SendMessageType = ReturnType<typeof sendMessage>
type RemoveMessageType = ReturnType<typeof removeMessage>
type ChangeNewMessageArea = ReturnType<typeof changeNewMessageArea>

//State
let initialState: MessagesStateType = {

    dialogs: [
        {id: 1, name: 'Alexandr'},
        {id: 2, name: 'Anvar'},
        {id: 3, name: 'Anna'},
        {id: 4, name: 'Igor'},
        {id: 5, name: 'Lena'},
    ],

    messages: [
        {id: 1, message: 'Hi!'},
        {id: 2, message: 'Hello!'},
        {id: 3, message: 'How are you?'},
        {id: 4, message: 'I`m fine. How ary you?'},
        {id: 5, message: 'I am OK.'},
        {id: 6, message: 'OK. Mebye go to the walk?'},
    ],

    newMessageArea: ''
}

//Reducer
export const messagesReducer = (state: MessagesStateType = initialState, action: MessagesActionType) => {
    switch (action.type) {

        case SEND_MESSAGE: {

            const allMessageID = [...state.messages.map(i=> i.id)]
            let newID = 1

            const generateID = (id: number): number => {
                return (allMessageID.some(i => +i === +id))
                    ? generateID(id+1)
                    : id
            }

            return {
                ...state,
                messages: [...state.messages, {
                    id: generateID(newID),
                    message: state.newMessageArea
                }],
                newMessageArea: ''
            }
        }

        case REMOVE_MESSAGE: {
            return {
                ...state,
                messages: state.messages.filter(i => i.id !== action.id)
            }
        }

        case CHANGE_NEW_MESSAGE_AREA: {
            return {
                ...state,
                newMessageArea: action.text
            }
        }

        default: {
            return state
        }
    }
}