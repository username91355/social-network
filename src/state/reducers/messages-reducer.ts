import {IUser, serverAPI} from "../../api/api";
import {ADD_TO_FRIENDS, REMOVE_FROM_FRIENDS, TAddFriend, TRemoveFriend} from "./users-reducer";
import {Nullable} from "./app-reducer";

//constants
const SET_FRIEND_LIST = 'socialNetwork/messagesReducer/SET_FRIEND_LIST';
const CHANGE_NEW_MESSAGE_AREA = 'socialNetwork/messagesReducer/CHANGE_NEW_MESSAGE_AREA';
const SEND_MESSAGE = 'socialNetwork/messagesReducer/SEND_MESSAGE';
const REMOVE_MESSAGE = 'socialNetwork/messagesReducer/REMOVE_MESSAGE';
const SET_ERROR = 'socialNetwork/messagesReducer/SET_ERROR';

//initialization state
export const iState = {

    friends: [],

    messages: {
        20345: [
            {id: 1, message: 'Hi!', outgoing: false},
            {id: 2, message: 'Hello!', outgoing: true},
            {id: 3, message: 'How are you?', outgoing: false},
            {id: 4, message: 'I`m fine. How ary you?', outgoing: true},
            {id: 5, message: 'I am OK.', outgoing: true},
            {id: 6, message: 'OK. Maybe go to the walk?', outgoing: false},
        ],
        19850: [
            {id: 1, message: 'Hi. How are you?', outgoing: true},
            {id: 2, message: 'Good afternoon! I\'m on a business trip in Norway', outgoing: false},
        ],
        2: [
            {id: 1, message: 'Hi. Thanks for the help.', outgoing: true},
            {id: 2, message: 'Hi. You are always welcome!', outgoing: false},
            {id: 3, message: 'How are you?', outgoing: false},
        ]
    },
    newMessageText: '',
    messagesError: ''
}

//reducer
export const messagesReducer = (state: IMessagesState = iState, action: TMessagesReducerActions) => {
    switch (action.type) {
        case SET_FRIEND_LIST:
            return {
                ...state,
                friends: action.list
            }
        case CHANGE_NEW_MESSAGE_AREA:
            return {
                ...state,
                newMessageText: action.value
            }
        case SEND_MESSAGE: {
            const allMessageID = [...state.messages[action.userId].map((p: any) => p.id)]
            const generateID = (id: number): number => {
                return (allMessageID.some(i => +i === +id))
                    ? generateID(id + 1)
                    : id
            }
            return {
                ...state,
                messages: {
                    ...state.messages,
                    [action.userId]: [
                        ...state.messages[action.userId],
                        {
                            id: generateID(Math.max(...allMessageID)),
                            message: state.newMessageText,
                            outgoing: true
                        }]
                },
                newMessageText: ''
            }
        }
        case REMOVE_MESSAGE:
            return {
                ...state,
                messages: {
                    ...state.messages,
                    [action.userId]: state.messages[action.userId].filter(m => m.id !== action.messageId)
                },
            }
        case ADD_TO_FRIENDS:
            return {
                ...state,
                messages: {
                    ...state.messages,
                    [action.userId]: []
                }
            }
        case REMOVE_FROM_FRIENDS:
            delete state.messages[action.userId]
            return {...state}
        case SET_ERROR:
            return {
                ...state,
                messagesError: action.error
            }
        default:
            return state
    }
}

//action creators
export const sendMessage = (userId: number) => ({type: SEND_MESSAGE, userId} as const)
export const removeMessage = (userId: number, messageId: number) => ({type: REMOVE_MESSAGE, userId, messageId} as const)
export const setFriendsList = (list: any) => ({type: SET_FRIEND_LIST, list} as const)
export const changeNewMessageArea = (value: string) => ({type: CHANGE_NEW_MESSAGE_AREA, value} as const)
export const setMessagesError = (error: string | null) => ({type: SET_ERROR, error} as const)

//thunks
export const setFriends = () => async (dispatch: any) => {
    dispatch(setMessagesError(null))
    const response = await serverAPI.getUsers(20, 1, '', true)

    if (!response.error) {
        dispatch(setFriendsList(response.items))
    } else {
        dispatch(setMessagesError(response.error))
    }
}

//types
interface IMessagesState {
    friends: IUser[]
    messages: {
        [key: number]: IMessage[]
    }
    newMessageText: string
    messagesError: Nullable<string>
}

export interface IMessage {
    id: number
    message: string
    outgoing: boolean
}

export type TMessagesReducerActions =
    TSendMessage
    | TRemoveMessage
    | TSetFriendsList
    | TChangeNewMessageArea
    | TSetMessagesError
    | TAddFriend
    | TRemoveFriend
type TSendMessage = ReturnType<typeof sendMessage>
type TRemoveMessage = ReturnType<typeof removeMessage>
type TSetFriendsList = ReturnType<typeof setFriendsList>
type TChangeNewMessageArea = ReturnType<typeof changeNewMessageArea>
type TSetMessagesError = ReturnType<typeof setMessagesError>