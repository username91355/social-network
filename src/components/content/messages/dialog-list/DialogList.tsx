import {NavLink} from "react-router-dom";
import styles from "./DialogList.module.css";
import React from "react";
import avatar from '../../../../assets/img/avatar-small.png'
import {TUser} from "../../../../redux/reducers/messages-reducer";

type TProps = {
    userList: Array<TUser>
}

const DialogList: React.FC<TProps> = props => {

    const {
        userList
    } = props

    return <div className={styles.dialogs__container}>
        {
            userList.map(d => {
                return <div key={d.id} className={styles.dialogs__item}>

                    <NavLink to={`/messages/${d.id}`}
                             className={({isActive}) => isActive
                                 ? styles.dialogs__link_active
                                 : styles.dialogs__link}>
                        <div className={styles.dialogs__link_info}
                             style={{display: 'flex', alignItems: 'center'}}>
                            <img src={d.photos.small || avatar} alt="avatar"/>
                            <div>
                                <h3>{d.name}</h3>
                                <div>{d.status || 'Status not set'}</div>
                            </div>
                        </div>
                    </NavLink>
                </div>
            })
        }
    </div>
}

export default DialogList