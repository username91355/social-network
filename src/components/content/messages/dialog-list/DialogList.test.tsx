import {TUser} from "../../../../redux/reducers/messages-reducer";
import {render, screen} from "@testing-library/react";
import DialogList from "./DialogList";
import {BrowserRouter} from "react-router-dom";

const testData: Array<TUser> = [
    {
        followed: true,
        id: 1,
        name: 'Alice',
        photos: {
            small: 'smallAlice',
            large: 'largeAlice'
        },
        status: null,
        uniqueUrlName: null
    },
    {
        followed: true,
        id: 2,
        name: 'Alex',
        photos: {
            small: 'smallAlex',
            large: 'largeAlex'
        },
        status: 'Status',
        uniqueUrlName: null
    }
]

const DialogListWithBrowserRouter = (testData: Array<TUser>) => {
    return <BrowserRouter>
        <DialogList userList={testData}/>
    </BrowserRouter>
}

describe('Dialog list component', () => {
    it('DialogList render', () => {
        render(DialogListWithBrowserRouter(testData));

        expect(screen.getByText('Alice')).toBeInTheDocument()
        expect(screen.getByText('Alex')).toBeInTheDocument()
        expect(screen.getAllByRole('img').length).toBe(2)
        expect(screen.getByText(/Status not set/i)).toBeInTheDocument()
        expect(screen.getByText('Status')).toBeInTheDocument()
    });

    it('Empty contact list snapshot', ()=> {
        const dialogList = render(DialogListWithBrowserRouter(testData));

        expect(dialogList).toMatchSnapshot()
    });
});