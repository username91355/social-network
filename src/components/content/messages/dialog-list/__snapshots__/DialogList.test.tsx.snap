// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog list component Empty contact list snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dialogs__container"
      >
        <div
          class="dialogs__item"
        >
          <a
            class="dialogs__link"
            href="/messages/1"
          >
            <div
              class="dialogs__link_info"
              style="display: flex; align-items: center;"
            >
              <img
                alt="avatar"
                src="smallAlice"
              />
              <div>
                <h3
                  class="dialogs__link_info_name"
                >
                  Alice
                </h3>
                <div>
                  Status not set
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="dialogs__item"
        >
          <a
            class="dialogs__link"
            href="/messages/2"
          >
            <div
              class="dialogs__link_info"
              style="display: flex; align-items: center;"
            >
              <img
                alt="avatar"
                src="smallAlex"
              />
              <div>
                <h3
                  class="dialogs__link_info_name"
                >
                  Alex
                </h3>
                <div>
                  Status
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dialogs__container"
    >
      <div
        class="dialogs__item"
      >
        <a
          class="dialogs__link"
          href="/messages/1"
        >
          <div
            class="dialogs__link_info"
            style="display: flex; align-items: center;"
          >
            <img
              alt="avatar"
              src="smallAlice"
            />
            <div>
              <h3
                class="dialogs__link_info_name"
              >
                Alice
              </h3>
              <div>
                Status not set
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="dialogs__item"
      >
        <a
          class="dialogs__link"
          href="/messages/2"
        >
          <div
            class="dialogs__link_info"
            style="display: flex; align-items: center;"
          >
            <img
              alt="avatar"
              src="smallAlex"
            />
            <div>
              <h3
                class="dialogs__link_info_name"
              >
                Alex
              </h3>
              <div>
                Status
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
